server/backend/cron/averageRatingComputation.go

```diff
@@ -0,0 +1,213 @@
+	CafeteriaID int32   `gorm:"column:cafeteriaID;foreignKey:cafeteriaID;type:int;" json:"cafeteriaID"`
+	DishID      int32   `gorm:"column:dishID;foreignKey:id;type:int;" json:"dishID"`
+	Average     float32 `json:"average"`
+	Min         int8    `json:"min"`
+	Max         int8    `json:"max"`
+	Std         float32 `json:"std"`
+}
+
+type averageCafeteriaTags struct {
+	CafeteriaID int32   `gorm:"column:cafeteriaID;foreignKey:cafeteriaID;type:int;" json:"cafeteriaID"`
+	TagID       int32   `gorm:"foreignKey:tagRatingID;column:tagID;type:int" json:"tagID"`
+	Average     float32 `json:"average"`
+	Min         int8    `json:"min"`
+	Max         int8    `json:"max"`
+	Std         float32 `json:"std"`
+}
+
+type averageDishTags struct {
+	CafeteriaID int32   `gorm:"column:cafeteriaID;foreignKey:cafeteriaID;type:int;" json:"cafeteriaID"`
+	TagID       int32   `gorm:"foreignKey:id;column:tagID;type:int" json:"tagID"`
+	DishID      int32   `gorm:"column:dishID;foreignKey:id;type:int;" json:"dishID"`
+	Average     float32 `json:"average"`
+	Min         int8    `json:"min"`
+	Max         int8    `json:"max"`
+	Std         float32 `json:"std"`
+}
+
+type averageDishNameTags struct {
+	CafeteriaID int32   `gorm:"column:cafeteriaID;foreignKey:cafeteriaID;type:int;" json:"cafeteriaID"`
+	TagID       int32   `gorm:"foreignKey:id;column:tagID;type:int" json:"tagID"`
+	Average     float32 `json:"average"`
+	Min         int8    `json:"min"`
+	Max         int8    `json:"max"`
+	Std         float32 `json:"std"`
+}
+
+/*
+This cronjob precomputes average ratings of all cafeteria ratings, dish ratings and all three types of tags.
+They are grouped (e.g. All Ratings for "Mensa_garching") and the computed values will then be stored in a table with the suffix "_result"
+*/
+func (c *CronService) averageRatingComputation() error {
+	computeAverageForCafeteria(c)
+	computeAverageForDishesInCafeterias(c)
+	computeAverageCafeteriaTags(c)
+	computeAverageForDishesInCafeteriasTags(c)
+	computeAverageNameTags(c)
+
+	return nil
+}
+
+func computeAverageNameTags(c *CronService) {
+	var results []averageDishNameTags
+	err := c.db.Raw("SELECT  mr.cafeteriaID as cafeteriaID, mnt.tagnameID as tagID, AVG(mnt.points) as average, MAX(mnt.points) as max, MIN(mnt.points) as min, STD(mnt.points) as std" +
+		" FROM dish_rating mr" +
+		" JOIN dish_name_tag mnt ON mr.dishRating = mnt.correspondingRating" +
+		" GROUP BY mr.cafeteriaID, mnt.tagnameID").Scan(&results).Error
+
+	if err != nil {
+		log.Println(err)
+	} else {
+		c.db.Where("1=1").Delete(&cafeteria_rating_models.DishNameTagAverage{})
+		for _, v := range results {
+			cafeteria := cafeteria_rating_models.DishNameTagAverage{
+				CafeteriaID: v.CafeteriaID,
+				Average:     v.Average,
+				TagID:       v.TagID,
+				Min:         v.Min,
+				Max:         v.Max,
+				Std:         v.Std,
+			}
+
+			errCreate := c.db.Model(&cafeteria_rating_models.DishNameTagAverage{}).Create(&cafeteria).Error
```

Comment:
```suggestion
			err := c.db.Model(&cafeteria_rating_models.DishNameTagAverage{}).Create(&cafeteria).Error
```

Can we also insert them in batch instead if one column at a time? Gorm should be able to do this:
```go
var rows []model.SomeRow
// fill array...
db.Insert(&rows)
```